src/System.Runtime.Numerics/tests/ComplexTests.cs

```diff
@@ -1360,15 +1360,37 @@ public static void Tan_Basic(double real, double imaginary)
 
         public static IEnumerable<object[]> Tan_Advanced_TestData()
         {
+
+            // .NET does not compute simple trig functions of large values correctly, so we can't make any
+            // assertions about complex trig functions with large real parts.
+
+            yield return new object[] { 0.0, double.MaxValue, 0.0, 1.0 };
+            yield return new object[] { 0.0, -double.MaxValue, 0.0, -1.0 };
+
+            yield return new object[] { 0.0, double.PositiveInfinity, 0.0, 1.0 };
+            yield return new object[] { 0.0, double.NegativeInfinity, 0.0, -1.0 };
+
+            yield return new object[] { 0.0, double.NaN, double.NaN, double.NaN };
+            yield return new object[] { double.NaN, 0.0, double.NaN, double.NaN };
+            yield return new object[] { double.NaN, double.PositiveInfinity, double.NaN, double.NaN };
+            yield return new object[] { double.NaN, double.NaN, double.NaN, double.NaN };
+
+            // Simple regression test; previous code returned wrong result for this argument
```

Comment:
You can leave off this comments; I think it just confuses things to talk about what "previous code" did.